linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1643,76 +1680,1188 @@ where
+    // Take one validator down
+    builder.set_fault_type([3], FaultType::Offline).await;
+
+    let blob0 = HashedBlob::test_blob("blob0");
+    let blob0_id = blob0.id();
+
+    // Publish blob on chain 1
+    let publish_certificate = client0.publish_blob(blob0).await.unwrap().unwrap();
+    assert!(publish_certificate
+        .value()
+        .executed_block()
+        .unwrap()
+        .requires_blob(&blob0_id));
+
+    builder
+        .set_fault_type([0, 1, 2], FaultType::DontSendConfirmVote)
+        .await;
+
+    client1.synchronize_from_validators().await.unwrap();
+    let blob1 = HashedBlob::test_blob("blob1");
+    let blob1_id = blob1.id();
+    let b0_result = client1
+        .reads_and_publishes_blobs(&[blob0_id], &[blob1])
+        .await;
+
+    assert!(b0_result.is_err());
+    assert!(client1.pending_block().is_some());
+
+    let validator_manager = builder
+        .node(0)
+        .chain_info_with_manager_values(chain_id2)
+        .await
+        .unwrap()
+        .manager;
+    assert!(validator_manager
+        .requested_locked
+        .clone()
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob0_id
+        })));
+    assert!(validator_manager
+        .requested_locked
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob1_id
+        })));
+
+    let validator_manager = builder
+        .node(1)
+        .chain_info_with_manager_values(chain_id2)
+        .await
+        .unwrap()
+        .manager;
+    assert!(validator_manager
+        .requested_locked
+        .clone()
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob0_id
+        })));
+    assert!(validator_manager
+        .requested_locked
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob1_id
+        })));
+
+    let validator_manager = builder
+        .node(2)
+        .chain_info_with_manager_values(chain_id2)
+        .await
+        .unwrap()
+        .manager;
+    assert!(validator_manager
+        .requested_locked
+        .clone()
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob0_id
+        })));
+    assert!(validator_manager
+        .requested_locked
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob1_id
+        })));
```

Comment:
And why not do these three in a loop?
(If we need them at all.)

Reply:
I think it's good to check, so will do in a loop. Happy to just remove also, if you think it's not needed